### ISSUE TITLE
Promethueus: trackQuery rudderstack event called multiple time in Scenes

### ISSUE THREAD

**Issue by bohandley** [grafana/grafana#77294]
Actions that could be taken:

- [x] Remove tracking (if it's not used) *edit, we use it all for UX data research
- ~~Only track from explore as intended~~
- [ ] Figure out why it tracks twice per request 
- [ ] If they are useful, do we need to send this to both intercom and rudderstack? * needs more research

In Scenes, the [rudderstack event to track a query](https://github.com/grafana/grafana/pull/61004) is called multiple times.

"...This service page triggers 6 metric queries, which triggers 12 intercom requests, and 14 rudderstack requests. most are with event_name "grafana_prometheus_query_executed". I think tracking every request seems too heavy, here we have 4x each metric request." [More context here if you are part of Grafana](https://raintank-corp.slack.com/archives/C04Q4ETKXHD/p1698394274979429)

![Image](https://github.com/grafana/grafana/assets/25674746/897ddca5-859c-4c30-bb02-23f8ba80d3f3)

To fix this issue, here are some suggestions that will need investigation:

1. Buffering the tracking - how could this be done?

2. Moving tracking to the backend - Can rudderstack events be sent from the backend?

3. Removing the trackQuery function from the query `tap` observable - Need to talk with UX about whether they have anymore plans to measure query frontend options with rudderstack events.

**Comment by bohandley** [comment]
This issue is outside the scope of metrics squad so we are removing our assignment from it but keeping the issue alive for future reference.

**Comment by jimkutz** [comment]
@bohandley how can I help with the research? Our Rudderstack costs have been increasing and `grafana_prometheus_query_executed` is our most expensive event (~$15k a year for just this event)

**Comment by bohandley** [comment]
@jimkutz These are the things (below) we need help with. As a developer, I was told that we just place a certain line of code and rudderstack sends the event and stores it. I don't think it is this simple. I am not sure why intercom is involved and our squad does not have the bandwidth currently to do a deep dive into rudderstack. Any help with these questions below would be nice.

- [ ] Figure out why it tracks twice per request 
- [ ] If they are useful, do we need to send this to both intercom and rudderstack? * needs more research

In Scenes, the [rudderstack event to track a query](https://github.com/grafana/grafana/pull/61004) is called multiple times.

"...This service page triggers 6 metric queries, which triggers 12 intercom requests, and 14 rudderstack requests. most are with event_name "grafana_prometheus_query_executed". I think tracking every request seems too heavy, here we have 4x each metric request." [More context here if you are part of Grafana](https://raintank-corp.slack.com/archives/C04Q4ETKXHD/p1698394274979429)

**Comment by jimkutz** [comment]
@shih-chris @jmarbach any idea why we're sending this to Intercom? @shih-chris I've seen a few queries against the `grafana_prometheus_query_executed` event. Not sure how accurate the event is based on the comments above.

**Comment by shih-chris** [comment]
## TLDR
1. I don't believe this is a primary driver for our rudderstack costs, and therefore not a huge priority (though happy to be proven wrong)
2. That said, I do think that these `grafana_prometheus_query_executed` events should be removed from the App O11y app, since they are not user-generated queries, but rather auto-generated ones that App O11y uses to operate
3. Intercom _seems_ to be an ops.grafana-ops.net issue, rather than one in production instances (I tested it on my own personal free instance and didn't see similar events)


## Details
> Remove tracking (if it's not used) *edit, we use it all for UX data research
> Only track from explore as intended

I definitely agree that the `grafana_prometheus_query_executed` event is helpful, particularly since we've seen that writing PromQL can be a hurdle for some new users as they onboard to Grafana.

However, in this case, I believe these are auto-generated PromQL queries (**that our App O11y app uses to run**), rather than queries that our end-users are creating.

For that reason, I'd personally think it makes sense to remove these `grafana_prometheus_query_executed` events.

That said, I am curious if we see this across ALL scenes apps? I suspect it's only App O11y. As such, **I don't think this change will have a huge impact on our rudderstack costs** (until App O11y adoption takes off 🚀 )

> Figure out why it tracks twice per request

No idea about this one. @ivanahuckova - I know you've worked with rudderstack events in the past. Have you run into this before and/or any guidance on where to look?

>  If they are useful, do we need to send this to both intercom and rudderstack? * needs more research

Haven't been able to look in detail here, but two thoughts:
1. all rudderstack events from hosted grafana are sent to Intercom, since Intercom is set up as a destination; however, I've not typically seen intercom events like this on my test instance
4. my gut is that the intercom issue is only present on ops.grafana-ops.net - we've seen weird intercom interactions on ops in the past

Definitely curious to get @jmarbach's take on the above!

**Comment by jmarbach** [comment]
I speculate one cause for the duplicate request to Intercom may be because of our connection settings from RudderStack to Intercom. We currently connect RudderStack and Intercom in "**Device mode**", which means that RudderStack loads the Intercom SDK directly on Grafana Cloud instances and the Intercom SDK sends events directly to Intercom.

Here's relevant documentation from RudderStack:

[Intercom Destination
](https://www.rudderstack.com/docs/destinations/streaming-destinations/intercom/)[How Device mode works
](https://www.rudderstack.com/docs/destinations/rudderstack-connection-modes/#how-device-mode-works)

A screenshot of the Intercom destination config page in RudderStack:
<img width="1884" alt="Screenshot 2024-02-20 at 10 07 10 PM" src="https://github.com/grafana/grafana/assets/935049/f57e6df3-19b3-400d-a7c8-e62dabe3b45b">

There are some reasons why Device mode is beneficial to us, namely latency between Grafana Cloud and Intercom. We can reconsider this connection and move to **Cloud mode** only  if we are able to still use Intercom effectively.

**Comment by itsmylife** [comment]
Closing this as this is not a case anymore after this PR https://github.com/grafana/grafana/pull/86054
